### PR TITLE
API: split pr for metrics protos

### PR DESCRIPTION
**Pull request by smonero**
### Description

It was requested to split this PR https://github.com/lyft/clutch/pull/2223/files to just the protos, so here they are :)

The idea here is that there will be a "metrics" service in Clutch that can get metrics from a store, such as m3 or something else. It follows a similar pattern as the `diagnostics` api in private. It could potentially "fan-out" to providers, as shown in the config proto. Something discussed at a meeting was that even though not necessarily all providers will have the same fields as prometheus does, it is ok for some of the fields to be blank.

### Testing Performed

none needed
### GitHub Issue

Fixes #

### CODE REVIEW

**Comment by smonero** [comment]
return type should be one:
provider style configuration
fanout to different providers?

enable in config section

example:
https://github.com/lyft/clutch-private/blob/main/backend/config/dash-config.yaml#L4-L9

topology uses data blob that it ingests

**Review by miguelmolina95 — APPROVED** [review]

File: api/metrics/v1/metrics.proto [hunk 1]
```diff
@@ -33,18 +38,22 @@ message GetMetricsRequest {
   repeated Query metric_queries = 1 [ (validate.rules).repeated = {min_items : 1} ];
 }
 
-message Metric {
+message MetricDataPoint {
```
Comment (danielhochman) [inline api/metrics/v1/metrics.proto:41]: nit: redundant with package name
```suggestion
message DataPoint {
```

**Review by danielhochman — APPROVED** [review]
small nits

File: api/config/service/metrics/v1/metrics.proto [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+package clutch.config.service.metrics.v1;
+
+option go_package = "github.com/lyft/clutch/backend/api/config/service/metrics/v1;metricsv1";
+
+import "validate/validate.proto";
+
+message MetricsProvider {
+  string api_endpoint = 1 [ (validate.rules).string = {min_bytes : 1} ];
+  int32 request_timeout = 2;
```
[the remark this answers is not in the document]
Comment (smonero) [reply api/config/service/metrics/v1/metrics.proto:11]: yeah idk now that I think about its not useful so I'll stick to the context deadline

**Pull request merged**